$OpenBSD: patch-config-scripts_cups-pdf_m4,v 1.2 2010/03/31 16:20:02 ajacoutot Exp $

Security fix for CVE-2009-3608, CVE-2009-3609.
CUPS "pdftops" Filter Data Handling Integer Overflow Vulnerabilities

Patch from Ubuntu: poppler-based-pdftops-fixes.dpatch

--- config-scripts/cups-pdf.m4.orig	Sat Aug  8 00:30:30 2009
+++ config-scripts/cups-pdf.m4	Thu Dec 10 15:28:50 2009
@@ -79,6 +79,12 @@ case "x$with_pdftops" in
 	;;
 esac
 
+if test "x$CUPS_PDFTOPS" != x; then
+	if `$CUPS_PDFTOPS -h 2>&1 | grep -q -- -origpagesizes 2>/dev/null`; then
+		AC_DEFINE(HAVE_PDFTOPS_WITH_ORIGPAGESIZES)
+	fi
+fi
+
 AC_DEFINE_UNQUOTED(CUPS_PDFTOPS, "$CUPS_PDFTOPS")
 AC_DEFINE_UNQUOTED(CUPS_GHOSTSCRIPT, "$CUPS_GHOSTSCRIPT")
 AC_SUBST(PDFTOPS)
